pages/starsearch.tsx

```diff
@@ -24,28 +24,107 @@ export async function getServerSideProps(context: GetServerSidePropsContext) {
+export default function StarSearchPage({ userId, bearerToken }: StarSearchPageProps) {
   const [starSearchState, setStarSearchState] = useState<"initial" | "chat">("initial");
+  const [chat, setChat] = useState<StarSearchChat[]>([]);
+  const [isRunning, setIsRunning] = useState(false);
+
+  const submitPrompt = async (prompt: string) => {
+    if (isRunning) {
+      return;
+    }
+    if (starSearchState === "initial") {
+      setStarSearchState("chat");
+    }
+    setIsRunning(true); // disables input
+
+    // add user prompt to history
+    setChat((history) => {
+      const temp = history;
+      temp.push({ author: "You", content: prompt });
+      return temp;
+    });
+
+    // get ReadableStream from API
+    const baseUrl = new URL(process.env.NEXT_PUBLIC_API_URL!);
+    const response = await fetch(`${baseUrl}/star-search/stream`, {
+      method: "POST",
+      body: JSON.stringify({
+        query_text: prompt,
+      }),
+      headers: {
+        Accept: "*/*",
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${bearerToken}`,
+      },
+    });
+
+    if (response.status !== 200) {
+      setChat((history) => {
+        const temp = history;
+        temp.push({ author: "StarSearch", content: "There's been an error. Try again." });
+        return temp;
+      });
+      setIsRunning(false); // enables input
+      return;
+    }
+
+    setChat((history) => {
+      const temp = history;
+      temp.push({ author: "StarSearch", content: "" });
+      return temp;
+    });
+
+    const decoder = new TextDecoderStream();
+    const reader = response.body?.pipeThrough(decoder).getReader();
+    while (true) {
+      const { done, value } = await reader!.read();
+      if (done) {
+        setIsRunning(false); // enables input
+        return;
+      }
+
+      const values = value.split("\n");
+      values
+        .filter((v) => v.startsWith("data:"))
+        .forEach((v) => {
+          let { data } = v.match(/data:(?<data>.*)/ms)?.groups || { data: "" };
+          const result = /(\s{1}[!"#$%&'()*+,\-./:;<=>?@[\]^_`{|}~])(\w*)/g.test(data) ? data.trimStart() : data;
+          const temp = [...chat];
+          const changed = temp.at(temp.length - 1);
+          changed!.content += result;
+          setChat(temp);
+        });
+    }
+  };
+
   const renderState = () => {
     switch (starSearchState) {
       case "initial":
         return <Header />;
       case "chat":
-        return <ChatHistory userId={userId} />;
+        return <ChatHistory userId={userId} chat={chat} />;
     }
   };
+
   return (
     <ProfileLayout>
       <div className="relative -mt-1.5 flex flex-col p-4 lg:p-8 justify-between items-center w-full h-full grow bg-slate-50">
         {renderState()}
-        <StarSearchInput onSubmit={() => setStarSearchState(starSearchState === "initial" ? "chat" : "initial")} />
+        <StarSearchInput isRunning={isRunning} onSubmitPrompt={(prompt) => submitPrompt(prompt)} />
```

Comment:
refactor (nit):  the anonymous arrow function isn't necessary.

```suggestion
        <StarSearchInput isRunning={isRunning} onSubmitPrompt={submitPrompt} />
```